### PR TITLE
Added support for new Arduino board packaging system

### PR DESCRIPTION
**Pull request by PeterMitrano**
The new arduino IDE uses a new location for boards.txt files, so I've added that to also be found. This allows Arturo to be used for Due's, ESPs, Intel Edisons, ect...

A bunch of trailing whitespace also removed... you should get your IDE to fix that.

### CODE REVIEW

File: ano/environment.py [hunk 1]
```diff
@@ -74,7 +74,9 @@ class Environment(dict):
     arduino_user_dir_guesses = [
         'libraries',
         os.path.expanduser("~/Documents/Arduino"),
-        os.path.expanduser("~/Arduino")
+        os.path.expanduser("~/Arduino"),
+        os.path.expanduser("~/.arduino15"),
+        os.path.expanduser("~/.arduino")
```
Comment (PeterMitrano) [inline ano/environment.py:79]: This is the first part of the significant change.
There may be other versions than .arduino15 but I can't confirm.

File: ano/environment.py [hunk 2]
```diff
@@ -167,8 +169,8 @@ def _find(self, key, items, places, human_name, join, multi, optional):
         places = map(os.path.expanduser, places)
 
         glob_places = itertools.chain.from_iterable(glob(p) for p in places)
-        
-        print 'Searching for', human_name, '...',
+
+        print 'Searching for', human_name, ' in ', places
```
Comment (PeterMitrano) [inline ano/environment.py:173]: More useful output? Yeh or Neh? I'll leave that up to you.

File: ano/environment.py [hunk 3]
```diff
@@ -215,7 +217,8 @@ def find_arduino_user_dir(self, key, dirname_parts, items=None, human_name=None,
         return self.find_dir(key, items, self.arduino_user_places(dirname_parts), human_name, multi=multi, optional=optional)
 
     def find_arduino_file(self, key, dirname_parts, items=None, human_name=None, multi=False):
-        return self.find_file(key, items, self.arduino_dist_places(dirname_parts), human_name, multi=multi)
+        places = self.arduino_user_places(dirname_parts) + (self.arduino_dist_places(dirname_parts))
+        return self.find_file(key, items, places, human_name, multi=multi)
```
Comment (PeterMitrano) [inline ano/environment.py:221]: This is the another part that actually matters. Basically just search more locations

File: ano/environment.py [hunk 4]
```diff
@@ -251,7 +254,7 @@ def board_models(self):
         # - hardware/arduino/{chipset}/boards.txt (Arduino 1.5.x, chipset like `avr`, `sam`)
         # - hardware/{platform}/boards.txt (MPIDE 0.xx, platform like `arduino`, `pic32`)
         # we should find and merge them all
-        boards_txts = self.find_arduino_file('boards.txt', ['hardware', '**'], 
+        boards_txts = self.find_arduino_file('boards.txt', ['**', 'hardware', '**'],
```
Comment (PeterMitrano) [inline ano/environment.py:257]: again, search more locations...

**Comment by scottdarch** [comment]
I'll give this a try tonight. Thanks Peter!

**Comment by PeterMitrano** [comment]
A word of caution... this doesn't actually mean all the new boards you can get will work. For instance, I'm trying to use the Due, and while this correctly finds the boards.txt entry for it, avr-g++ still doesn't understand all the flags. My guess is that the arduino IDE provides some other toolchain that I don't know about... But I still think this is useful for boards that do use avr-g++

**Comment by scottdarch** [comment]
Yep. The mainline of Arturo is at end-of-life because of this. The original authors wrote their own compiler commands instead of using the ones in platform.txt which means it only works with avr architectures.

**Comment by PeterMitrano** [comment]
Oh cool, I didn't even notice platform.txt...
What's the status of the develop branch? I wouldn't mind helping out, since I've relied pretty heavily on arturo for a lot of projects. If you have some specific action items you're working I could possibly help out. Especially if it means I can compile for due ;)

**Comment by scottdarch** [comment]
The development branch is about 40% done. I've had some job changes recently so I haven't had time to work much on it. It's a total rewrite and it will generate MUCH better makefiles. The last place I left off was trying to get the build recipes right for building binaries. I believe the compilation step is working. I'd LOVE, LOVE, LOVE for someone else to dive in! Beyond all the makefile stuff the Python code has the potential to become a general python library for Arduino development allowing other projects to use it as the basis for, say, a new IDE. Let me know how I can help you get started if you have time.

**Comment by PeterMitrano** [comment]
Ok, well I'm still an undergrad so I have a more stable schedule. I'll
checkout the dev branch and try to see where it's at.
When you say become more than a build tool, my first thought is vim plugin
because python :)

Anyways,  I'd like to see this project stay alive so I'm willing to help if
you have time to get me introduced to it.

File: ano/environment.py (hunk 4 above)
Comment (scottdarch) [inline ano/environment.py:257]: I'm missing something with this change: '**' is already in the array. How is this different?

Comment (PeterMitrano) [inline ano/environment.py:257]: hmm so I believe these are appended and globbed, so `['hardware','**']` only searches `/hardware/**`, whereas `['**', 'hardware', '**']` would match anything with hardware somewhere in the path. such as `packages/esp8266/hardware/esp8266/2.0.0/boards.txt`

Comment (scottdarch) [inline ano/environment.py:257]: Ah, I see it now. Down in guess_at_places.

**Comment by scottdarch** [comment]
Thanks.

**Comment by PeterMitrano** [comment]
BTW--A friend pointed me to PlatformIO, which literally supports everything. I'm not sure if Arturo has any advantages, but I'm switching to PlatformIO for now. You should check it out! The CLI is good, and they have a very polished Atom extension as well.

**Pull request merged**